File: client/scalaz/src/main/scala/org/openapitools/client/api/SendigProduct.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SendigProduct._

case class SendigProduct (
  name: Option[String],
label: Option[String],
description: Option[String],
source: Option[String],
effectiveDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
links: Option[SendigProductLinks],
classes: Option[List[SendigClass]])

object SendigProduct {
  import DateTimeCodecs._

  implicit val SendigProductCodecJson: CodecJson[SendigProduct] = CodecJson.derive[SendigProduct]
  implicit val SendigProductDecoder: EntityDecoder[SendigProduct] = jsonOf[SendigProduct]
  implicit val SendigProductEncoder: EntityEncoder[SendigProduct] = jsonEncoderOf[SendigProduct]
}

File: client/clojure/src/cdisc_library_api/specs/sendig_product.clj
(ns cdisc-library-api.specs.sendig-product
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.sendig-product-links :refer :all]
            [cdisc-library-api.specs.sendig-class :refer :all]
            )
  (:import (java.io File)))


(def sendig-product-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :source) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :_links) sendig-product-links-spec
   (ds/opt :classes) (s/coll-of sendig-class-spec)
   })

(def sendig-product-spec
  (ds/spec
    {:name ::sendig-product
     :spec sendig-product-data}))
